src/ncar-bufr2nc-fortran/define_types_mod.f90

```diff
@@ -0,0 +1,371 @@
+         obtype     = 'aircraft'
+      case (522, 523)
+         obtype     = 'sfc' ! 'ship'
+      case (531, 532, 561, 562, 534)
+         obtype     = 'sfc' ! 'buoy'
+      case (511, 514, 540) ! mesonet 540
+         ! kx case (181, 281)
+         obtype     = 'sfc' ! 'synop'
+      case (512)
+         ! kx case (187, 287)
+         obtype     = 'sfc' ! 'metar'
+      case (63)
+         ! kx case (242:246, 252:253, 255)
+         obtype     = 'satwind'
+      case (581, 582, 583, 584)
+         ! ERS 581, QuikSCAT 582, WindSat 583, ASCAT 584
+         obtype     = 'ascat'
+      !case (74)
+      !   obtype     = 'gpspw'
+      !case (71, 73, 75, 76, 77)
+      ! t29=73/kx=229 (Wind profiler originating in PIBAL bulletins (tropical and European)
+      ! t29=77/kx=126 (Multi-Agency Profiler (MAP) RASS temperatures)
+      case (71, 73, 75, 76)
+         obtype     = 'profiler'
+      !case (571, 65)
+      !   obtype = 'ssmir' ! ssmi retrieval
+   end select
+
+end subroutine set_obtype_conv
+
+subroutine set_name_satellite(satid, satellite)
+
+! https://www.emc.ncep.noaa.gov/BUFRLIB/tables/CodeFlag_0_STDv33_LOC7.html#001007
+
+! assign satellite name based on BUFR SAID (0-01-007)
+
+   implicit none
+
+   integer(i_kind),  intent(in)  :: satid
+   character(len=*), intent(out) :: satellite
+
+   satellite = 'unknown'
+
+   select case ( satid )
+      case (   3 ); satellite = 'metop-b'
+      case (   4 ); satellite = 'metop-a'
+      case (   5 ); satellite = 'metop-c'
+      case ( 206 ); satellite = 'n15'
+      case ( 207 ); satellite = 'n16'
+      case ( 208 ); satellite = 'n17'
+      case ( 209 ); satellite = 'n18'
+      case ( 223 ); satellite = 'n19'
+      case ( 224 ); satellite = 'npp'
+      case ( 225 ); satellite = 'n20'
+      case ( 226 ); satellite = 'n21'
+      case ( 783 ); satellite = 'terra'
+      case ( 784 ); satellite = 'aqua'
+   end select
+
+end subroutine set_name_satellite
+
+subroutine set_name_sensor(instid, sensor)
+
+! https://www.emc.ncep.noaa.gov/BUFRLIB/tables/CodeFlag_0_STDv33_LOC7.html#002019
+
+! assign sensor name based on BUFR SIID (0-02-019) Satellite instruments
+
+   implicit none
+
+   integer(i_kind),  intent(in)  :: instid
+   character(len=*), intent(out) :: sensor
+
+   sensor = 'unknown'
+
+   select case ( instid )
+      case ( 570 ); sensor = 'amsua'
+      case ( 574 ); sensor = 'amsub'
+      case ( 203 ); sensor = 'mhs'
+      case ( 617 ); sensor = 'abi'
+      case ( 297 ); sensor = 'ahi'
+      case ( 207 ); sensor = 'seviri'
+      case ( 420 ); sensor = 'airs'
+      case ( 620 ); sensor = 'cris'
+      case ( 221 ); sensor = 'iasi'
+      case ( 389 ); sensor = 'modis'
+      case ( 616 ); sensor = 'viirs'
+   end select
+
+end subroutine set_name_sensor
+
+subroutine set_brit_obserr(name_inst, ichan, obserr)
+
+! set brightness temperature observation errors
+
+! For now it is a temporary subroutine to assign AMSU-A and MHS observation errors
```

Comment:
Assuming the permanent solution is to read in an input file, not writing the errors in the source explicitly.